server/id/id.go

```diff
@@ -70,6 +69,13 @@ func (alloc *AllocatorImpl) Alloc() (uint64, error) {
 
 // Generate synchronizes and generates id range.
 func (alloc *AllocatorImpl) Generate() error {
+	alloc.mu.Lock()
+	defer alloc.mu.Unlock()
+
+	return alloc.generateLocked()
+}
+
+func (alloc *AllocatorImpl) generateLocked() error {
```

Comment:
```suggestion
func (alloc *AllocatorImpl) unlockedGenerate() error {
```

Reply:
`xxxLocked` seems to be the common name in this project, also in `protobuf` or `goleveldb`.

Reply:
Also used in the official package such as [sync.Map](https://golang.org/src/sync/map.go).